en/developer_guide/best_practice_js.rst

```diff
@@ -0,0 +1,288 @@
+
+    function foo() {
+      return true;
+    }  // no semicolon here.
+
+
+Standards Feature:
+------------------
+
+Always preferred over non-standards features
+
+For maximum portability and compatibility, always prefer standards features over non-standards
+features (e.g., string.charAt(3) over string[3], ...).
+
+Wrapper objects and primitive types:
+------------------------------------
+
+Always preferred use primitive types.
+
+.. code-block:: js
+
+    // var string1 = String("Hello world");
+    var string1 = “Hello world”;
+    var string2 = new String(“Hello world”);
+
+    string1 === string2 // return false
+    string1 === string2.valueOf() // return true
+    string1.valueOf() === string2.valueOf() // return true
+
+    typeof String(0) == 'string';
+    typeof new String(0) == 'object';
+
+
+This is very useful for casting things to number, string and boolean.
+
+See : `Distinction between string primitives and String objects`_ by Mozilla.org.
+
+For-in Loop:
+------------
+
+Only for iterating over keys in an object/map/hash.
+Always use normal for loops when using arrays.
+
+.. code-block:: js
+    function printArrayForIn(arr) {
+      for (var key in arr) {
+        print(arr[key]);
+      }
+    }
+    function printArray(arr) {
+      var l = arr.length;
+      for (var i = 0; i < l; i++) {
+        print(arr[i]);
+      }
+    }
+
+    printArrayForIn([0,1,2,3]);  // This works.
+    printArray([0,1,2,3]);       // This works.
+
+    var a = new Array(10);
+    printArrayForIn(a);    // This is wrong.
+    printArray(a);         // This works.
+
+
+
+
+Tips :
+------
+Don't Use new Object()
+Use {} instead of new Object()
+Use "" instead of new String()
+Use 0 instead of new Number()
+Use false instead of new Boolean()
+Use [] instead of new Array()
+Use /()/ instead of new RegExp()
+Use function (){} instead of new function()
+
+.. code-block:: js
+
+    var x1 = {};           // new object
+    var x2 = "";           // new primitive string
+    var x3 = 0;            // new primitive number
+    var x4 = false;        // new primitive boolean
+    var x5 = [];           // new array object
+    var x6 = /()/;         // new regexp object
+    var x7 = function(){}; // new function object
+
+Function declarations within blocks:
+------------------------------------
+
+Do not do this:
+
+.. code-block:: js
+
+    if (x) {
+      function foo() {}
+    }
+
+While most script engines support Function Declarations within blocks it is not part of
+ECMAScript (see ECMA-262, clause 13 and 14). Worse implementations are inconsistent with
+each other and with future EcmaScript proposals. ECMAScript only allows for Function
+Declarations in the root statement list of a script or function. Instead use a variable
+initialized with a Function Expression to define a function within a block:
+
+Namespace:
+----------
+
+The first step to good JavaScript object management is having a namespace, or a JavaScript
```

Comment:
If it's the first step to good js maybe this should be mentionned earlier rather than flooded in the middle of the doc